internal/errs2/ignore.go

```diff
@@ -12,20 +12,29 @@ import (
 )
 
 // IgnoreCanceled returns nil, when the operation was about canceling.
-func IgnoreCanceled(originalError error) error {
-	err := originalError
-	for err != nil {
-		if err == context.Canceled ||
+func IgnoreCanceled(err error) error {
+	if IsFunc(err, func(err error) bool {
+		return err == context.Canceled ||
 			err == grpc.ErrServerStopped ||
-			err == http.ErrServerClosed {
-			return nil
+			err == http.ErrServerClosed
+	}) {
+		return nil
+	}
+
+	return err
+}
+
+// IsFunc checks whether any of the underlying errors matches the func
+func IsFunc(err error, is func(err error) bool) bool {
+	if err == nil {
+		return is(err)
+	}
+	for {
+		if is(err) {
+			return true
 		}
-		unwrapped := errs.Unwrap(err)
-		if unwrapped == err {
-			return originalError
+		if err = errs.Unwrap(err); err == nil {
```

Comment:
the exit condition for the for loop should be err == errs.Unwrap(err). errs.Unwrap(err) returns err if err can't be unwrapped, not nil